.circleci/config.yml

```diff
@@ -312,6 +313,75 @@ jobs:
     <<: *test-defaults
     environment:
       - TEST_TARGET=sanity
+  build-waterfall:
+    <<: *defaults
+    working_directory: ~/
+    steps:
+      - checkout:
+          path: emscripten/
+      - run:
+          name: install package dependencies
+          command: |
+            apt-get update -q
+            apt-get install -q -y python3 cmake
+      - run:
+          name: install emsdk
+          command: |
+            wget https://github.com/juj/emsdk/archive/master.tar.gz
+            tar -xf master.tar.gz
+            cd emsdk-master
+            ./emsdk --notty update-tags
+            export WATERFALL_LKGR=`python -c "print [x.split('\"')[3] for x in open('upstream/lkgr.json').readlines() if '\"build\"' in x][0]"`
+            echo "LKGR: $WATERFALL_LKGR"
+            ./emsdk --notty install node-8.9.1-64bit # TODO: should emsdk install this automatically?
+            ./emsdk --notty install clang-upstream-$WATERFALL_LKGR-64bit
+            ./emsdk --notty activate clang-upstream-$WATERFALL_LKGR-64bit
+            cd -
+            # Use LLVM, clang, lld binaries from the emsdk
+            export LLVM="$HOME/emsdk-master/upstream/$WATERFALL_LKGR/bin"
+            # Remove the emsdk version of emscripten to save space in the
+            # persistent workspace and to avoid any confusion with that version
+            # we are trying to test.
+            rm -r ~/emsdk-master/upstream/$WATERFALL_LKGR/emscripten/
+            # Use Binaryen from the emsdk's waterfall (wasm backend can't use ports)
```

Comment:
More-specifically we can't use Binaryen as a port, ports-in-general should still work

Reply:
Good point, this was very unclear, fixing.